server/controllers/publicProfiles.js

```diff
@@ -0,0 +1,58 @@
+const asyncHandler = require("express-async-handler");
+const List = require("../models/List");
+const Product = require("../models/Product");
+
+exports.getPublicLists = asyncHandler(async (req, res, next) => {
+  const lists = await List.find({ isPrivate: false }).select([
+    "-creator",
+    "-userIds",
+  ]);
+  res.status(200).json({
+    success: true,
+    lists: lists,
+  });
+});
+
+exports.getListProducts = asyncHandler(async (req, res, next) => {
+  const listId = req.params.listId;
+
+  if (!listId) {
+    res.status(400).json({
+      error: { message: "listId unspecified" },
+    });
+    return;
+  }
+
+  try {
+    const list = await List.findById(listId);
+
+    if(!list) {
+      res.status(404).json({
+        error: { message: "Not Found" },
+      });
+      return;
+    }
+  
+    if(list.isPrivate == true) {
+      res.status(401).json({
+        error: { message: "unauthorized" },
+      });
+      return;
+    }
+  
+    const products = await Product.find().where('_id').in(list.products);
+  
+    res.status(200).json({
+      success: {
+        message: "Found",
+        products: products,
+      },
+    });
```

Comment:
Could we just populate the list products and send it directly off the list?

Reply:
I've thought about it in task description there should be another router to fetch the list products
https://app.zenhub.com/workspaces/engg-60b5066927ea0b000ebf6adb/issues/hatchways/team-athens/25